mirgecom/flux.py

```diff
@@ -72,6 +72,7 @@ def lfr_flux(q_tpair, flux_func, normal, lam):
         object array of :class:`meshmode.dof_array.DOFArray` with the
         Lax-Friedrichs/Rusanov flux.
     """
-    flux_avg = 0.5*(flux_func(q_tpair.int)
-                    + flux_func(q_tpair.ext))
-    return flux_avg @ normal - 0.5*lam*(q_tpair.ext - q_tpair.int)
+    flux_avg = 0.5*(flux_func(cv_tpair.int)
+                    + flux_func(cv_tpair.ext))
+    return flux_avg.join() @ normal - 0.5*lam*(cv_tpair.ext.join()
+                                               - cv_tpair.int.join())
```

Comment:
These joins are after:

```python
return flux_avg @ normal - 0.5*lam*(cv_tpair.ext - cv_tpair.int)
```

gave this error:

```
E ValueError: matmul: Input operand 0 does not have enough dimensions (has 0, gufunc core with signature (n?,k),(k,m?)->(n?,m?) requires 1)
```

After using the "join" of flux_avg:

```python
return flux_avg.join() @ normal - 0.5*lam*(cv_tpair.ext - cv_tpair.int)
```

this error:

```
E TypeError: unsupported operand type(s) for *: 'DOFArray' and 'ConservedVars'
```